cfme/storage/volume_backup.py

```diff
@@ -0,0 +1,252 @@
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_volume_backup and
+            self.title.text == 'Cloud Volume Backups')
+
+
+class VolumeBackupDetailsView(VolumeBackupView):
+    """The detail Volume Backup page"""
+    @property
+    def is_displayed(self):
+        expected_title = '{} (Summary)'.format(self.context['object'].name)
+
+        return (
+            self.title.text == expected_title and
+            self.entities.breadcrumb.active_location == expected_title)
+
+    toolbar = View.nested(VolumeBackupDetailsToolbar)
+    sidebar = View.nested(VolumeBackupDetailSidebar)
+    entities = View.nested(VolumeBackupDetailsEntities)
+
+
+class VolumeRestoreView(VolumeBackupView):
+    """The restore Volume backup view"""
+    @property
+    def is_displayed(self):
+        return False
+
+    volume_name = BootstrapSelect(name='volume_id')
+    save = Button('Save')
+    reset = Button('Reset')
+    cancel = Button('cancel')
+
+
+@attr.s
+class VolumeBackup(BaseEntity, WidgetasticTaggable):
+    """ Model of an Storage Volume Backups in cfme
+
+    Args:
+        name: name of the backup
+        provider: provider
+    """
+    name = attr.ib()
+    provider = attr.ib()
+
+    def restore(self, name):
+        """Restore the volume backup. this feature included in 5.9 and above.
+
+        Args:
+            name: volume name
+        """
+        view = navigate_to(self, 'Restore')
+        view.volume_name.fill(name)
+        view.save.click()
+        view.flash.assert_success_message('Restoring Cloud Volume "{}"'.format(self.name))
+
+    def refresh(self):
+        self.provider.refresh_provider_relationships()
+        self.browser.refresh()
+
+    @property
+    def size(self):
+        """ size of cloud volume backup.
+
+        Returns:
+            :py:class:`int' size of volume backup.
+        """
+        view = navigate_to(self, 'Details')
+        return view.entities.properties.get_text_of('Size')
+
+    @property
+    def status(self):
+        """ Present status of cloud volume backup.
+
+        Returns:
+            :py:class:`str' status [available/restoring] of volume backup.
+        """
+        view = navigate_to(self, 'Details')
+        return view.entities.properties.get_text_of('Status')
+
+    @property
+    def volume(self):
+        """ volume name of backup.
+
+        Returns:
+            :py:class:`str' respective volume name.
+        """
+        view = navigate_to(self, 'Details')
+        return view.entities.relationships.get_text_of('Cloud Volume')
+
+
+@attr.s
+class VolumeBackupCollection(BaseCollection):
+    """Collection object for :py:class:'cfme.storage.volume_backups.VolumeBackup' """
+    ENTITY = VolumeBackup
+
+    def all(self):
+        """returning all backup objects"""
+        view = navigate_to(self, 'All')
+        backups = [self.instantiate(name=item, provider=self.filters.get('provider'))
```

Comment:
**REQUIRED:** This doesn't filter by provider from what I can see. ```self.filters.get('provider')``` will return to you the provider that the collection is filtered with, but you would then need a conditional on the end of the list comprehension to only supply the backups that are related to that provider.